src/ixdat/backends/memory_backend.py

```diff
@@ -1,45 +1,44 @@
-"""This module implements the simplest backend, which just counts rows in memory"""
+from .backend_base import BackendBase
 
 
-class BackendBase:
-    """Base class listing the functions that must be implemented in a database backend.
+class MemoryBackend(BackendBase):
+    """A backend that assigns id's and keeps track of objects for later access.
 
-    A backend defines where and how all Saveable objects (inheriting from the Saveable
-    class) save their data. This is a key part of the seamless interoperability of
-    ixdat classes and experimental databases. Each Saveable class roughly corresponds
-    to a table, and the save() and get() functions correspond to inserting and
-    selecting from a database table.
-
-    Backends inheriting from this base class are in modules of the ixdat.backends folder
+    This means that a Saveable object can make a serializable representation of itself
+    that includes the identity in the memory backend of objects that it references, and
+    a new Saveable object made from this representation can find the original objects.
+    See db.Saveable.as_dict(), which ensures that "child objects" are in memory.
     """
 
-    name = "memory"
+    backend_type = "memory"
+    address = "here"
 
     def __init__(self):
         """Initialize the backend with dict for {table_name (str): id_counter (int)}"""
-        self.next_available_ids = {}
+        super().__init__()
+        self.objects = {}
 
-    def get_next_available_id(self, table_name):
+    def get_next_available_id(self, table_name, obj=None):
         """Return the id counter for table_name, starting with 1."""
-        if table_name in self.next_available_ids:
-            i = self.next_available_ids[table_name]
-            self.next_available_ids[table_name] += 1
-        else:
-            i = 1
-            self.next_available_ids[table_name] = 2
+        i = super().get_next_available_id(table_name)
+        if table_name not in self.objects:
```

Comment:
right... I guess when I was writing this I was thinking that `MemoryBackend` is unique in that it would never assign an id to something it doesn't "save", which for it just means "remember". But thinking about it now that's completely backwards. `save` should call `get_next_available_id` just like any other ixdat backend that can't outsource it's id generation.

Reply:
fixed. Now  `get_next_available_id` does not save, but is called by `save`.